src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -626,28 +631,48 @@ export class TransactionApi implements ITransactionApi {
         ...args,
       });
       const url = `${this.baseUrl}/api/v1/safes/${args.safeAddress}/multisig-transactions/`;
-      return await this.dataSource.get({
-        cacheDir,
-        url,
-        notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
-        networkRequest: {
-          params: {
-            safe: args.safeAddress,
-            ordering: args.ordering,
-            executed: args.executed,
-            trusted: args.trusted,
-            execution_date__gte: args.executionDateGte,
-            execution_date__lte: args.executionDateLte,
-            to: args.to,
-            value: args.value,
-            nonce: args.nonce,
-            nonce__gte: args.nonceGte,
-            limit: args.limit,
-            offset: args.offset,
+      return await this.dataSource
+        .get<Page<MultisigTransaction>>({
+          cacheDir,
+          url,
+          notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
+          networkRequest: {
+            params: {
+              safe: args.safeAddress,
+              ordering: args.ordering,
+              executed: args.executed,
+              trusted: args.trusted,
+              execution_date__gte: args.executionDateGte,
+              execution_date__lte: args.executionDateLte,
+              to: args.to,
+              value: args.value,
+              nonce: args.nonce,
+              nonce__gte: args.nonceGte,
+              limit: args.limit,
+              offset: args.offset,
+            },
           },
-        },
-        expireTimeSeconds: this.defaultExpirationTimeInSeconds,
-      });
+          expireTimeSeconds: this.defaultExpirationTimeInSeconds,
+        })
+        .then(async (data): Promise<Page<MultisigTransaction>> => {
+          const hasConfirmationsRequired = data.results.every((tx) => {
+            return tx.confirmationsRequired !== null;
+          });
+          if (hasConfirmationsRequired) {
+            return data;
+          }
+
+          const results = await Promise.all(
+            data.results.map(async (tx) => {
+              return await this._setConfirmationsRequired(tx);
+            }),
+          );
```

Comment:
Nit: this iterates `data.results` twice when there is an item with `confirmationsRequired === null`, and also executes `_setConfirmationsRequired(tx)` for all the transactions in the page.

No strong requirement, but maybe we can optimize it a bit by:
```javascript
        .then(
          async (data): Promise<Page<MultisigTransaction>> => ({
            ...data,
            results: await Promise.all(
              data.results.map(async (tx) =>
                tx.confirmationsRequired !== null
                  ? tx
                  : await this._setConfirmationsRequired(tx),
              ),
            ),
          }),
        );
```

This would also simplify the `_setConfirmationRequired` implementation because the initial check can be removed (and the input type) as the function would only receive transactions affected by the bug:

And a final adjustment could be also done in the other `_setConfirmationRequired` call in line 705:
```javascript
        .then(async (tx) =>
          transaction.confirmationsRequired !== null
            ? tx
            : await this._setConfirmationsRequired(tx),
        );
```

Again, this is just a nit, feel free to keep the current implementation if you think is clearer!

Reply:
I adjusted it in all three instances in 09995bb3079d9c9be8f8cd0a5a654c1cf3295963.